fastestimator/op/numpyop/univariate/rua.py

```diff
@@ -409,35 +279,98 @@ def __init__(self,
+            AutoContrast(inputs=inputs, outputs=outputs, mode=mode),
+            Equalize(inputs=inputs, outputs=outputs, mode=mode),
+            Posterize(inputs=inputs, outputs=outputs, mode=mode, num_bits=7),
+            Solarize(inputs=inputs, outputs=outputs, mode=mode, threshold=256),
+            Sharpness(inputs=inputs, outputs=outputs, mode=mode, limit=0.9),
+            Contrast(inputs=inputs, outputs=outputs, mode=mode, limit=0.9),
+            Color(inputs=inputs, outputs=outputs, mode=mode, limit=0.9),
+            Brightness(inputs=inputs, outputs=outputs, mode=mode, limit=0.9),
+            ShearX(inputs=inputs, outputs=outputs, mode=mode, shear_coef=0.5),
+            ShearY(inputs=inputs, outputs=outputs, mode=mode, shear_coef=0.5),
+            TranslateX(inputs=inputs, outputs=outputs, mode=mode, shift_limit=0.33),
+            TranslateY(inputs=inputs, outputs=outputs, mode=mode, shift_limit=0.33)
         ]
-        return aug_options
+        aug_options = self._parse_aug_choices(magnitude_coef=(level / 30.), choices=to_list(choices))
+
+        # Calculating number of augmentation to apply at each training iteration
+        N_min = 1
+        N_max = min(len(aug_options), 5)
+        N = level * (N_max - N_min) / 30 + N_min
+        N_guarantee, N_p = int(N), N % 1
+
+        self.ops = [OneOf(*aug_options) for _ in range(N_guarantee)]
+        if N_p > 0:
+            self.ops.append(Sometimes(OneOf(*aug_options), prob=N_p))
+
+    def _parse_aug_choices(self, magnitude_coef: float, choices: List[Union[str, NumpyOp]]) -> List[NumpyOp]:
+        """Parse the augmentation choices to determine the final list of augmentations to apply.
+
+        Args:
+            magnitude_coef: Factor to set the range for magnitude of augmentation.
+            choices: list of augmentations to apply.
+
+        Returns:
+            List of augmentations to apply.
+
+        Raises:
+            AssertionError: If augmentations to add and remove are mixed.
+            AttributeError: If augmentation choices don't have a 'set_rua_level' method.
+            ValueError: If 'defaults' is provided with augmentation strings to add or remove, or wrong names provided.
+        """
+        custom_ops = [op for op in choices if not isinstance(op, str)]
+        remove_ops = [op for op in choices if isinstance(op, str) and op.startswith("!")]
+        add_ops = [op for op in choices if isinstance(op, str) and not (op.startswith("!") or (op == "defaults"))]
+
+        assert len(remove_ops)==0 or len(add_ops)==0, \
+            "Either add or remove ops, not both, found {} and {}".format(add_ops, remove_ops)
+
+        if len(remove_ops) > 0:
+            if "defaults" in choices:
+                raise ValueError("Can't provide 'defaults' value with ops to remove, found: {}".format(remove_ops))
+            aug_list = self.default_aug_list
+            aug_names = [op.__class__.__name__ for op in aug_list]
```

Comment:
This would probably be simplified if self.aug_list was a dictionary rather than a list